convert2rhel/actions/conversion/set_efi_config.py

```diff
@@ -0,0 +1,219 @@
+# Copyright(C) 2024 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import logging
+import os
+import shutil
+
+from convert2rhel import actions, grub, systeminfo
+from convert2rhel.grub import CENTOS_EFIDIR_CANONICAL_PATH, RHEL_EFIDIR_CANONICAL_PATH
+
+
+logger = logging.getLogger(__name__)
+
+
+class NewDefaultEfiBin(actions.Action):
+    id = "NEW_DEFAULT_EFI_BIN"
+
+    def run(self):
+        """Check that the expected RHEL UEFI binaries exist."""
+        super(NewDefaultEfiBin, self).run()
+
+        if not grub.is_efi():
+            logger.info("BIOS detected. Nothing to do.")
+            return
+
+        new_default_efibin = None
+        for filename in grub.DEFAULT_INSTALLED_EFIBIN_FILENAMES:
+            efi_path = os.path.join(RHEL_EFIDIR_CANONICAL_PATH, filename)
+            if os.path.exists(efi_path):
+                logger.info("UEFI binary found: %s" % efi_path)
+                new_default_efibin = efi_path
+                break
+            logger.debug("UEFI binary %s not found. Checking next possibility..." % efi_path)
+        if not new_default_efibin:
+            self.set_result(
+                level="ERROR",
+                id="RHEL_UEFI_BINARIES_DO_NOT_EXIST",
+                title="RHEL UEFI binaries do not exist",
```

Comment:
They could exist, but we just couldn't find them
```suggestion
                title="RHEL UEFI binaries not found",
```